pandas/core/arrays/boolean.py

```diff
@@ -706,7 +710,7 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
             return libmissing.NA
 
         # if we have numeric op that would result in an int, coerce to int if possible
-        if name in ["sum", "prod"] and notna(result):
+        if name == "prod" and notna(result):
```

Comment:
you are going to add a mask_ops for this, yes?

Reply:
also for min/max

Reply:
I am planning to do the same for prod (and probably min/max), yes, but would prefer to keep this PR focussed on sum